src/catalog/catalog_storage_manager.cpp

```diff
@@ -0,0 +1,117 @@
+
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// catalog_storage_manager.cpp
+//
+// Identification: src/catalog/catalog_storage_manager.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+#include "catalog/catalog_storage_manager.h"
+
+#include <iostream>
+
+#include "catalog/database_metrics_catalog.h"
+#include "catalog/manager.h"
+#include "catalog/query_metrics_catalog.h"
+#include "catalog/table_metrics_catalog.h"
+#include "catalog/index_metrics_catalog.h"
+#include "common/exception.h"
+#include "common/macros.h"
+#include "expression/date_functions.h"
+#include "expression/string_functions.h"
+#include "expression/decimal_functions.h"
+#include "index/index_factory.h"
+#include "util/string_util.h"
+
+namespace peloton {
+namespace catalog {
+
+// Get instance of the global catalog storage manager
+CatalogStorageManager *CatalogStorageManager::GetInstance(void) {
+  static std::unique_ptr<CatalogStorageManager> global_catalog_storage_manager(
+      new CatalogStorageManager());
+  return global_catalog_storage_manager.get();
+}
+
+CatalogStorageManager::CatalogStorageManager() {}
+
+//===--------------------------------------------------------------------===//
+// GET WITH OID - DIRECTLY GET FROM STORAGE LAYER
+//===--------------------------------------------------------------------===//
+
+/* Find a database using its oid from storage layer,
+ * throw exception if not exists
+ * */
+storage::Database *CatalogStorageManager::GetDatabaseWithOid(
```

Comment:
Do you mind double checking that whenever other methods invoke Catalog Storage Manager's methods (like GetDatabaseWithOid()), it's always surrounded by try-catch block?